### DISCUSSION TITLE
Bundle Size

### DISCUSSION THREAD

**Discussion by L-TheG** [wevm/wagmi#866] [General]
Hi there,

I don't mean to put any pressure, or demand anything, I am just curious and couldn't really find anything recent about it:
Are there any possibilities or plans of reducing the bundle size of the initial load, if the main page of an app does not require wagmi? The first load is pretty hefty after all.

Thanks :)

**Comment by johnbchron** [comment]
This is still an issue. Seems to add almost half a megabyte (when uncompressed) to the `_app` chunk. Does it need the full weight of the `ethers` library, or can some of it be excluded? I tried the targeted imports below but it didn't seem to help much, and there doesn't seem to be a good way to lazy-load:
```js
import { createClient, configureChains, WagmiConfig } from 'wagmi';
import { publicProvider } from 'wagmi/providers/public';
import { polygon } from 'wagmi/chains'
```

**Reply by jxom** [reply]
We have some news coming in the next fortnight that indirectly relates to this. ;)

**Reply by lightninglu10** [reply]
Any updates on this @jxom ? been about 2 fortnights.

**Reply by jxom** [reply]
https://twitter.com/wagmi_sh/status/1632814459235500032?s=20

wagmi v1 will be porting from ethers.js to viem, which will result in ~3x lower bundle size – we are working on this right now.

**Reply by lightninglu10** [reply]
Ah cool, is there an ETA for this release, or even a beta? Would be good to have on our radar

**Reply by jxom** [reply]
See more in this discussion: [The Future of wagmi](https://github.com/wagmi-dev/wagmi/discussions/2207)
